### PR TITLE
Quickfix for ponds migration

### LINKED ISSUE #53423: When loading a save from a previous version 'invalid overmap special id'
### Describe the bug

When loading a save from a previous version, it gives several errors.
The world is also from the previous version.
P.S. The errors in the screenshots are repeated 3-4 times.

### Steps to reproduce

1. Create a world and save in the previous version;
2. Update the game;
3. Load the game.

### Expected behavior

The game does not show these errors.

### Versions and configuration

8b7a449 (2021-12-13-0951)

### Additional context

[debug.log](https://github.com/CleverRaven/Cataclysm-DDA/files/7705223/debug.log)

### PR DESCRIPTION
**Pull request by Night-Pryanik**
#### Summary
None

#### Purpose of change
Fixes #53423.

#### Describe the solution
Obsoleted overmap specials and applied migration.

#### Describe alternatives you've considered
None.

#### Testing
Loaded a save from pre-changes-to-ponds version without errors popping up.

#### Additional context
None.

### CODE REVIEW

**Comment by Maleclypse** [comment]
Im on my phone and couldn’t see why the required build failed. So I kicked the tests.

**Comment by Night-Pryanik** [comment]
Build is complaining on absence of t_console_broken terrain, which was converted to furniture long ago, but my change doesn't touch neither terrain nor furniture.

**Comment by kevingranade** [comment]
This is failing because you moved the obsoletion files, there is some ordering bug in data loading that we haven't been able to track down.  If you want to start moving entries into obsoletions/whatever, feel free to start putting new entries there, but don't mass-migrate existing entries.

File: data/json/obsolete_terrains.json [hunk 1]
```diff
@@ -24,7 +24,9 @@
       "ants_larvae",
       "ants_larvae_acid",
       "ants_queen",
-      "ants_queen_acid"
+      "ants_queen_acid",
+      "pond_forest",
+      "pond_swamp"
```
Comment (ZhilkinSerg) [inline data/json/obsolete_terrains.json:29]: Obsolete overmap terrain ids with direction should also be here - `pond_forest_east`, `pond_swamp_north` etc. (see https://github.com/CleverRaven/Cataclysm-DDA/issues/61543).

**Pull request merged**